p2p/transport/webrtc/listener.go

```diff
@@ -329,8 +329,8 @@ func (l *listener) Multiaddr() ma.Multiaddr {
 func addOnConnectionStateChangeCallback(pc *webrtc.PeerConnection) <-chan error {
 	errC := make(chan error, 1)
 	var once sync.Once
-	pc.OnConnectionStateChange(func(state webrtc.PeerConnectionState) {
-		switch state {
+	pc.OnConnectionStateChange(func(_ webrtc.PeerConnectionState) {
+		switch pc.ConnectionState() {
```

Comment:
This works fine, but the change makes it better in case there are ordering issues for the callback. The callback does need to guarantee that we can access connection state.

Reply:
Apparently it doesn't work fine. https://github.com/pion/webrtc/pull/2702#issuecomment-2156006877

So we do need this change.